core/src/nucleus/ribosome/callback/validation_package.rs

```diff
@@ -88,6 +88,44 @@ pub fn get_validation_package_definition(
                 Some(call.parameters.into_bytes()),
             )?
         }
+        EntryType::LinkRemove => {
+            let link_remove = match entry {
+                Entry::LinkRemove(link_remove) => link_remove,
+                _ => {
+                    return Err(HolochainError::ValidationFailed(
+                        "Failed to extract LinkRemove".into(),
+                    ));
+                }
+            };
+            let (base, target) = links_utils::get_link_entries(link_remove.link(), &context)?;
+
+            let link_definition_path = links_utils::find_link_definition_in_dna(
+                &base.entry_type(),
+                link_remove.link().tag(),
+                &target.entry_type(),
+                &context,
+            )?;
+
+            let wasm = context
+                .get_wasm(&link_definition_path.zome_name)
+                .expect("Couldn't get WASM for zome");
+
+            let params = LinkValidationPackageArgs {
+                entry_type: link_definition_path.entry_type_name,
+                tag: link_definition_path.tag,
+                direction: link_definition_path.direction,
+            };
+
+            let call = ZomeFnCall::new("", None, "__hdk_get_validation_package_for_link", params);
+
+            ribosome::run_dna(
+                &dna.name.clone(),
+                context,
+                wasm.code.clone(),
+                &call,
+                Some(call.parameters.into_bytes()),
+            )?
+        }
```

Comment:
Except for the first match, this block is exactly the same as above for `LinkAdd`, right? Can we DRY this out?

Reply:
We could but I think this could be a follow up! I had some ideas about how we could DRY out the whole workflow